spec/ParseUser.spec.js

```diff
@@ -3448,40 +3448,123 @@ describe('Parse.User testing', () => {
       });
   });
 
-  it('should not allow updates to hidden fields', done => {
+  it('should not allow updates to hidden fields', async () => {
     const emailAdapter = {
       sendVerificationEmail: () => {},
       sendPasswordResetEmail: () => Promise.resolve(),
       sendMail: () => Promise.resolve(),
     };
-
     const user = new Parse.User();
     user.set({
       username: 'hello',
       password: 'world',
       email: 'test@email.com',
     });
+    await reconfigureServer({
+      appName: 'unused',
+      verifyUserEmails: true,
+      emailAdapter: emailAdapter,
+      publicServerURL: 'http://localhost:8378/1',
+    });
+    await user.signUp();
+    user.set('_email_verify_token', 'bad', { ignoreValidation: true });
+    await expectAsync(user.save()).toBeRejectedWith(
+      new Parse.Error(Parse.Error.INVALID_KEY_NAME, 'Invalid field name: _email_verify_token.')
+    );
+  });
 
-    reconfigureServer({
+  it('should allow updates to fields with maintenanceKey', async () => {
+    const emailAdapter = {
+      sendVerificationEmail: () => {},
+      sendPasswordResetEmail: () => Promise.resolve(),
+      sendMail: () => Promise.resolve(),
+    };
+    const user = new Parse.User();
+    user.set({
+      username: 'hello',
+      password: 'world',
+      email: 'test@email.com',
+    });
+    await reconfigureServer({
       appName: 'unused',
+      maintenanceKey: 'test2',
```

Comment:
What happens if the `maintenanceKey` is not set in the server config? I think if no key is set it should not be possible to access internal fields. Contrary to that, I think that the server refuses to start if on `masterKey` is provided.

Reply:
```js
if (req.config.maintenanceKey && info.maintenanceKey === req.config.maintenanceKey) {
```

If no maintenance key is set, it's not possible to perform requests from the client side with maintenance auth. Parse Server will still use maintenance key internally to update maintenance fields